core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/PathPropertiesImpl.kt

```diff
@@ -74,10 +74,17 @@ data class PathPropertiesImpl(
                 infra.getRoutesOnTrackChunk(dirChunkId).filter { route ->
                     pathRoutes!!.contains(route)
                 }
-            assert(routeOnChunk.size < 2) {
-                "a train cannot follow more than one route at the same time"
-            }
-            val route = routeOnChunk.singleOrNull()?.let { routeId -> infra.getRouteName(routeId) }
+            // Warning: Technically, in the following situation, the path would loop, and you could
+            // have 2 itineraries
+            // in the path with the same commonChunk, with no way to know the true speed limit. For
+            // now, we take the
+            // first itinerary's speed limit.
+            // -> - -
+            //        \
+            //         end
+            //           \
+            // - start - - - commonChunk - ->
+            val route = routeOnChunk.firstOrNull()?.let { routeId -> infra.getRouteName(routeId) }
```

Comment:
You could add a `TODO: add a warning` in the comment. It's not really useful for now, but we can grep for it when we'll have an actual warning system in place.